terraform_compliance/extensions/terraform.py

```diff
@@ -573,3 +571,36 @@ def is_type(self, resource, mode):
             return resource['address'].split('.')[0] == mode
 
         return False
+
+    def process_module_calls(self, module_resource, parents_modules=None):
+        if parents_modules is None:
+            parents_modules = []
+
+        resources = []
+        for k, v in module_resource.items():
+            # Set the naming correct (for cases like module.a.module.b.module.c...)
+            current_module_level = parents_modules
+            current_module_level.append('module.{}'.format(k))
+            module_name = ".".join(current_module_level)
```

Comment:
Module naming is slightly incorrect here. Line 584 joins module_names that are on the same level.

For example:
```
module_resource = {
    'parent_module_1': { ... }
    'parent_module_2': {
       'module': {
            'resources': {
                'child_resource_1': {}
            }
        }
     }
}
```
The address of child_resource_1 would be `module.parent_module_1.parent_module_2.name_of_child_resource_1`
instead of `module.parent_module_2.name_of_child_resource_1`

Reply:
ah yes, of course. Fixing and adding a similar scenario to unit tests as well.

Reply:
Fixed in 7e91670.

python ..